public/editor-client/src/aiText/index.ts

```diff
@@ -0,0 +1,103 @@
+import { request } from "../api";
+import { Conversation, RequestOptions, Role } from "../types/AiText";
+import { getConfig } from "../config";
+
+export const doAiRequest = async (
+  res: (value: string) => void,
+  rej: (error: string) => void,
+  data: { prompt: string; instruction?: string }
+): Promise<void> => {
+  const config = getConfig();
+
+  if (!config) {
+    throw new Error("Invalid __BRZ_PLUGIN_ENV__");
+  }
+
+  try {
+    const {prompt, instruction} = data;
+    const conversation: Conversation[] = [];
+
+    switch (instruction) {
+      case "extend":
+        conversation.push({
+          role: Role.User,
+          content: "Expand the following text by 2 times: ",
+        });
+        conversation.push({
+          role: Role.Assistant,
+          content: `${prompt}`,
+        });
+        break;
+      case "shorten":
+        conversation.push({
+          role: Role.User,
+          content: "Make the following text shorter: ",
+        });
+        conversation.push({
+          role: Role.Assistant,
+          content: `${prompt}`,
+        });
+        break;
+      case "simplify":
+        conversation.push({
+          role: Role.User,
+          content: "Simplify the language of following text: ",
+        });
+        conversation.push({
+          role: Role.Assistant,
+          content: `${prompt}`,
+        });
+        break;
+      case "English":
+      case "German":
+      case "French":
+      case "Spanish":
+      case "Italian":
+      case "Dutch":
+      case "Portuguese":
+      case "Polish":
+        conversation.push({
+          role: Role.Assistant,
+          content: `Translate text to ${instruction}: ${prompt}`,
+        });
+        break;
+      case "Professional":
+      case "Casual":
+      case "Confident":
+      case "Friendly":
+        conversation.push({
+          role: Role.User,
+          content: `Change the following text tone to '${instruction}': ${prompt}`,
+        });
+        break;
+      default:
+        conversation.push({
+          role: Role.Assistant,
+          content: prompt,
+        });
+        break;
+    }
```

Comment:
https://github.com/bagrinsergiu/blox-editor/pull/23123#discussion_r1338072171

Reply:
Changed naming from instructions to instruction because in each request we ask for AI to do only one thing. We not need to do much things in 1 request

Reply:
talked with  @Nessunskill  and renamed to action because it makes more sense in our context